Content.Shared/_RMC14/Xenonids/Acid/SharedXenoAcidSystem.cs

```diff
@@ -99,16 +118,27 @@ public override void Update(float frameTime)
         if (_net.IsClient)
             return;
 
-        var query = EntityQueryEnumerator<CorrodingComponent>();
         var time = _timing.CurTime;
 
-        while (query.MoveNext(out var uid, out var corroding))
+        var damageableCorrodingQuery = EntityQueryEnumerator<DamageableCorrodingComponent>();
+        while (damageableCorrodingQuery.MoveNext(out var uid, out var damageableCorrodingComponent))
+        {
+            if (time > damageableCorrodingComponent.LastDamagedAt.Add(TimeSpan.FromSeconds(_corrosiveAcidTickDelaySeconds)))
+            {
+                DamageSpecifier damage = new(_prototypeManager.Index<DamageTypePrototype>(_corrosiveAcidDamageType), damageableCorrodingComponent.Dps * _corrosiveAcidTickDelaySeconds);
+                _damageable.TryChangeDamage(uid, damage, true);
+                damageableCorrodingComponent.LastDamagedAt = time;
+            }
+        }
```

Comment:
Ok I remember now, this is the part that can be pulled out outside the while
_prototypeManager.Index<DamageTypePrototype>(_corrosiveAcidDamageType);
You could even cache the prototype itself but then you would need to re-cache it if the cvar changes or prototypes are reloaded, but doing it once per update loop is fine
```suggestion
        var damageType = _prototypeManager.Index<DamageTypePrototype>(_corrosiveAcidDamageType);
        var damageableCorrodingQuery = EntityQueryEnumerator<DamageableCorrodingComponent>();
        while (damageableCorrodingQuery.MoveNext(out var uid, out var damageableCorrodingComponent))
        {
            if (time > damageableCorrodingComponent.LastDamagedAt.Add(TimeSpan.FromSeconds(_corrosiveAcidTickDelaySeconds)))
            {
                DamageSpecifier damage = new(damageType, damageableCorrodingComponent.Dps * _corrosiveAcidTickDelaySeconds);
                _damageable.TryChangeDamage(uid, damage, true);
                damageableCorrodingComponent.LastDamagedAt = time;
            }
        }
```

Reply:
shouldnt it store the DamageSpecifier too to not reallocate it every time

Reply:
Yes, wasn't sure if it was safe since some things might store the damage specifier but that needs to be done too

Reply:
Not sure how can I store damage specifier since it can be different for different barricades we are iterating over. Dps depends on the type of the acid applied. Unless we store damage specifier on the component itself instead of dps value that is. Or something along these lines.

Reply:
Which in turn means I'll have to iterate over all components and update damageSpecifiers for them if CVar changes.

Reply:
you can at the very least cache the DamageSpecifier instance to avoid reallocating it

Reply:
Done: 5db57f8130696a6e5fc20f2108b3902c68fd7863

I also think DamageableCorrodingComponent does not need to be repliacted over the network too. It's fully server side, client doesn't do anything with that data. It's not that much data being transfered, but every bit helps. Pelase correct me if I'm wrong and I'll revert the commit: 8f6e706ba6bef4c9dbce1545a58b6e5e3e4999be

Reply:
In that case you would want to move the compone to Server, but I would keep it networked so that the client can predict something breaking from the damage if necessary in the future

Reply:
Actually deletions don't run on the client, so keep it non networked but move it to Server

Reply:
Fair. I'll just revert it then. Which is done now.

Reply:
> Actually deletions don't run on the client, so keep it non networked but move it to Server

I believe that will require quite a bit stuff moved from Shared to Server since Shared code doesn't have access to Server code. Is that what we want to do?

Reply:
So, bottom line for now - I'll keep it as is, networked and in shared code.